### ISSUE TITLE
Can't invoke custom formatter with --require and --format 

### ISSUE THREAD

**Issue by dhempy** [bbatsov/rubocop#1584]
I suspect this is not a rubocop issue, but rather a user error on my part.  Sorry if I'm posting this in the wrong place.

You can see my fork at https://github.com/dhempy/rubocop/pull/1/files 

I'm writing a custom formatter, `quiet`.  I've tried invoking it two ways, as described at http://www.rubydoc.info/gems/rubocop/RuboCop/Formatter/BaseFormatter : 

`rubocop --require ~/dev/rubocop/lib/rubocop/formatter/quiet_formatter.rb --format QuietFormatter`
and
`rubocop --require ~/dev/rubocop/lib/rubocop/formatter/quiet_formatter.rb --format quiet`

I've confirmed the --require option is working, as it fails when I give it a bad filename.

When I give --format the class name QuietFormatter in the cli, I get `uninitialized constant QuietFormatter`: (I get the same error with any mixed-case formatter name)

```
$ rubocop --require ~/dev/rubocop/lib/rubocop/formatter/quiet_formatter.rb --format QuietFormatter
warning: parser/current is loading parser/ruby21, which recognizes
warning: 2.1.5-compliant syntax, but you are running 2.1.4.
uninitialized constant QuietFormatter
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/lib/rubocop/formatter/formatter_set.rb:79:in `const_get'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/lib/rubocop/formatter/formatter_set.rb:79:in `block in custom_formatter_class'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/lib/rubocop/formatter/formatter_set.rb:78:in `each'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/lib/rubocop/formatter/formatter_set.rb:78:in `reduce'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/lib/rubocop/formatter/formatter_set.rb:78:in `custom_formatter_class'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/lib/rubocop/formatter/formatter_set.rb:37:in `add_formatter'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/lib/rubocop/runner.rb:149:in `block in formatter_set'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/lib/rubocop/runner.rb:148:in `each'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/lib/rubocop/runner.rb:148:in `formatter_set'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/lib/rubocop/runner.rb:23:in `run'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/lib/rubocop/cli.rb:24:in `run'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/bin/rubocop:13:in `block in <top (required)>'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/2.1.0/benchmark.rb:294:in `realtime'
/Users/dhempy/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rubocop-0.28.0/bin/rubocop:12:in `<top (required)>'
/Users/dhempy/.rbenv/versions/2.1.4/bin/rubocop:23:in `load'
/Users/dhempy/.rbenv/versions/2.1.4/bin/rubocop:23:in `<main>'
```

FWIW, I also tried using the key (`quiet`) that I added to BUILTIN_FORMATTERS_FOR_KEYS in my fork and got, `No formatter for "quiet"` ... but I'm sure that is not relevant since my fork is not actually installed on my system as a gem.  I get the same thing if I use any lower case name.

My formatter is a clone of simple_text_formatter, and starts like this:

```
# encoding: utf-8

require 'rubocop/formatter/colorizable'
require 'rubocop/formatter/text_util'

module RuboCop
  module Formatter
    # A basic formatter that displays only files with offenses.
    # Offenses are displayed at compact form - just the
    # location of the problem and the associated message.
    # Differs from simple in that no output is displayed if there are no offenses.
    class QuietFormatter < BaseFormatter
      include Colorizable, PathUtil, TextUtil

  <snip>
```

I've run out of ideas and have not found anything similar reported in github issues, stackexchange, or Google.  Any idea what am I missing?

**Comment by jonas054** [comment]
A custom formatter doesn't need to be in the `RuboCop::Formatter` namespace, but if it is, you must specify it on the command line, so:

```
rubocop -r ~/.../quiet_formatter.rb -f RuboCop::Formatter::QuietFormatter
```

The other way to solve this would be to move out of the namespace:
```ruby
class QuietFormatter < RuboCop::Formatter::BaseFormatter
  include RuboCop::Formatter::Colorizable
  include RuboCop::PathUtil
  include RuboCop::Formatter::TextUtil
  # ...
end
```

**Comment by dhempy** [comment]
Perfect!  Thanks, @jonas054 !

**Comment by gamesover** [comment]
I can include `include RuboCop::Formatter::Colorizable` as the above example without any issue. But I cannot include `RuboCop::Cop::RangeHelp`. It will display error
